scipy/stats/_rvs_sampling.py

```diff
@@ -40,132 +39,12 @@ def rvs_ratio_uniforms(pdf, umax, vmin, vmax, size=1, c=0, random_state=None):
-    normalization factor of the density.
-
-    >>> f = lambda x: np.exp(-x**2 / 2)
-    >>> v_bound = np.sqrt(f(np.sqrt(2))) * np.sqrt(2)
-    >>> umax, vmin, vmax = np.sqrt(f(0)), -v_bound, v_bound
-    >>> rvs = stats.rvs_ratio_uniforms(f, umax, vmin, vmax, size=2500,
-    ...                                random_state=rng)
-
-    The K-S test confirms that the random variates are indeed normally
-    distributed (normality is not rejected at 5% significance level):
-
-    >>> stats.kstest(rvs, 'norm')[1]
-    0.250634764150542
-
-    The exponential distribution provides another example where the bounding
-    rectangle can be determined explicitly.
-
-    >>> rvs = stats.rvs_ratio_uniforms(lambda x: np.exp(-x), umax=1,
-    ...                                vmin=0, vmax=2*np.exp(-1), size=1000,
-    ...                                random_state=rng)
-    >>> stats.kstest(rvs, 'expon')[1]
-    0.21121052054580314
-
+    Please refer to `scipy.stats.sampling.RatioUniforms` for the documentation
+    and use ``RatioUniforms`` instead of this function which is deprecated.
     """
-    if vmin >= vmax:
-        raise ValueError("vmin must be smaller than vmax.")
-
-    if umax <= 0:
-        raise ValueError("umax must be positive.")
-
-    size1d = tuple(np.atleast_1d(size))
-    N = np.prod(size1d)  # number of rvs needed, reshape upon return
-
-    # start sampling using ratio of uniforms method
-    rng = check_random_state(random_state)
-    x = np.zeros(N)
-    simulated, i = 0, 1
-
-    # loop until N rvs have been generated: expected runtime is finite.
-    # to avoid infinite loop, raise exception if not a single rv has been
-    # generated after 50000 tries. even if the expected numer of iterations
-    # is 1000, the probability of this event is (1-1/1000)**50000
-    # which is of order 10e-22
-    while simulated < N:
-        k = N - simulated
-        # simulate uniform rvs on [0, umax] and [vmin, vmax]
-        u1 = umax * rng.uniform(size=k)
-        v1 = rng.uniform(vmin, vmax, size=k)
-        # apply rejection method
-        rvs = v1 / u1 + c
-        accept = (u1**2 <= pdf(rvs))
-        num_accept = np.sum(accept)
-        if num_accept > 0:
-            x[simulated:(simulated + num_accept)] = rvs[accept]
-            simulated += num_accept
-
-        if (simulated == 0) and (i*N >= 50000):
-            msg = ("Not a single random variate could be generated in {} "
-                   "attempts. The ratio of uniforms method does not appear "
-                   "to work for the provided parameters. Please check the "
-                   "pdf and the bounds.".format(i*N))
-            raise RuntimeError(msg)
-        i += 1
-
-    return np.reshape(x, size1d)
+    warnings.warn("Please use `RatioUniforms` from the "
+                  "`scipy.stats.sampling` namespace. The "
+                  "`scipy.stats.rvs_ratio_uniforms` namespace is deprecated.",
```

Comment:
We should give a deadline so the timeline is clear
```suggestion
                  "`scipy.stats.rvs_ratio_uniforms` namespace is deprecated"
                  "and will be removed in SciPy 1.14.0.",
```